query/src/main/scala/filodb/query/exec/PromQlRemoteExec.scala

```diff
@@ -45,12 +45,24 @@ case class PromQlRemoteExec(queryEndpoint: String,
 
   override def sendHttpRequest(execPlan2Span: Span, httpTimeoutMs: Long)
                               (implicit sched: Scheduler): Future[QueryResponse] = {
+
+    import PromCirceSupport._
+    import io.circe.parser
     remoteExecHttpClient.httpGet(queryContext.plannerParams.applicationId, queryEndpoint,
       requestTimeoutMs, queryContext.submitTime, getUrlParams())
       .map { response =>
-        response.unsafeBody match {
-          case Left(error) => QueryError(queryContext.queryId, error.error)
-          case Right(successResponse) => toQueryResponse(successResponse.data, queryContext.queryId, execPlan2Span)
+        if (response.body.isLeft)
```

Comment:
NiT: Open Parentheses in same line as `if` condition. Also please add a comment on why we check `.isLeft`